repo/object/object_writer.go

```diff
@@ -121,27 +123,19 @@ func (w *objectWriter) flushBuffer() error {
 	return nil
 }
 
-func (w *objectWriter) maybeCompressedContentBytes() (data []byte, isCompressed bool, err error) {
-	if w.compressor != nil {
-		compressedBytes, err := w.compressor.Compress(w.buffer.Bytes())
+func maybeCompressedContentBytes(comp compression.Compressor, b []byte) (data []byte, isCompressed bool, err error) {
+	if comp != nil {
+		compressedBytes, err := comp.Compress(b)
 		if err != nil {
 			return nil, false, errors.Wrap(err, "compression error")
 		}
 
-		if len(compressedBytes) < w.buffer.Len() {
+		if len(compressedBytes) < len(b) {
```

Comment:
Was this corrupting the stored data?

Reply:
yes, but only for longer `I` objects. short ones were fine.